#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/OsbWood.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

WorldInfo {
  info [
    "In this example, the robot moves down a slope."
    "The altimeter is used to record the robots height."
  ]
  title "Altimeter"
  basicTimeStep 8
  contactProperties [
    ContactProperties {
      material2 "body"
      coulombFriction [
        0
      ]
      softCFM 0.1
    }
  ]
}
Viewpoint {
  orientation -0.16747741127523166 -0.0670383327442875 0.9835940110916765 4.334005473412958
  position 0.24884199196179196 2.0576912254960864 0.49225638163151453
  follow "MyBot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
PointLight {
  attenuation 0 0 1
  intensity 0.6
  location 0.6 0 0
}
DEF SLOPE Solid {
  translation 0 0 0.15
  rotation 0 1 0 -0.4
  children [
    Shape {
      appearance OsbWood {
        textureTransform TextureTransform {
          scale 0.5 2
        }
      }
      geometry DEF BOX0 Box {
        size 0.8 0.2 0.02
      }
    }
  ]
  name "green box"
  boundingObject USE BOX0
}
Robot {
  translation 0.20483 0.0426693 0.247124
  rotation 0.14353942634221165 0.00829414473582794 -0.9896098424371345 3.026293667343932
  children [
    DEF smile Pose {
      translation 0.036 0 0.025
      rotation 0.25056300542381216 -0.9351130202419243 -0.25056300542381216 1.63783
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
    DEF ds0 DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 0 0 -1 -0.499
      children [
        DEF sensor Pose {
          rotation 0.5778648646508888 0.5770962987628834 -0.5770892999841452 -2.09394
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
    Altimeter {
    }
    DEF ds1 DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation 0 0 -1 -0.499
      children [
        DEF sensor Pose {
          rotation 0.5778648646508888 0.5770962987628834 -0.5770892999841452 -2.09394
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
    Pose {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -20.063169637844744
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.044999999999999686 0.025
        rotation -0.7014158531232039 0.7014158871035109 0.12661577431838011 3.3934839362328844
        children [
          DEF WHEEL Pose {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 98.30208719465305
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.04500000000000002 0.025
        rotation 0.6873557538701489 -0.6873558083790767 0.23469993674721185 3.602647766679766
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
  ]
  name "MyBot"
  contactMaterial "body"
  boundingObject Pose {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "my_controller"
}
